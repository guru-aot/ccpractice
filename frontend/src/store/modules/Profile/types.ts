export interface ProfileState {
  currentUser: ProfileModel[];
  loading?: boolean;
  successStatus?: boolean;
  errorStatus?: boolean;
}

export interface ProfileModel {
  genre: string;
}
