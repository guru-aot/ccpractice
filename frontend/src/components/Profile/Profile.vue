/**
 * 
 * Add books
 */

<template>
  <v-card class="mx-auto" max-width="600">
    <v-toolbar flat>
      <v-toolbar-title class="grey--text">My Preferences</v-toolbar-title>
      <div class="flex-grow-1"></div>
    </v-toolbar>
    <!-- {{userProfile}} -->
    <v-divider></v-divider>
    <v-alert type="success" v-if="successStatus">Profile Updated succesfully</v-alert>
    <v-alert type="error" v-if="errorStatus">Something went wrong...</v-alert>
    <v-card-text>
      <v-form>
        <v-container>
          <v-row>
            <v-col cols="12">
              <h1
                class="display-2 font-weight-bold mb-3"
              >Hello {{userProfile && userProfile.last_name}}</h1>
            </v-col>
          </v-row>
        </v-container>
      </v-form>
    </v-card-text>
  </v-card>
</template>
<script lang="ts">
import { Component, Vue, Watch } from 'vue-property-decorator';

import { Getter, namespace } from 'vuex-class';
const BookModule = namespace('BookModule');
const ProfileModule = namespace('ProfileModule');

@Component
export default class Profile extends Vue {
  @ProfileModule.Getter('userProfile') public userProfile!: [];
  @ProfileModule.Getter('successStatus') public successStatus!: boolean;
  @ProfileModule.Getter('errorStatus') public errorStatus!: boolean;

  /**
   * @description loading  genre on load
   */
  private mounted() {
    // this.$store.dispatch('AuthorModule/loadAuthors');
    this.$store.dispatch('ProfileModule/loadUserProfile');
  }
}
</script>
