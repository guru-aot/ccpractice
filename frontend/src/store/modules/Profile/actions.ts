import { ActionTree } from 'vuex';
import { ProfileState, ProfileModel } from './types';
import { RootState } from '../../types';
import axios from '@/lib/axios';
import { BASE_URL } from '@/config/urlList';

/**
 * profile
 *
 */
export const actions: ActionTree<ProfileState, RootState> = {
  /**
   * load users from server and set to store
   * @param {*} { commit }
   */
  // loadUserProfile({ commit }) {
  //   commit('SET_LOADING', true);
  //   axios
  //     .get(`${BASE_URL}${PROFILE_URL}`)
  //     .then(r => r.data)
  //     .then(users => {
  //       commit('SET_USERS_PROFILE', users);
  //       commit('SET_LOADING', false);
  //     });
  // },
  /**
   * clear all alerts
   * @param {*} { commit }
   */
  clearAlert({ commit }) {
    commit('SET_LOADING', false);
    commit('SET_PROFILE_SUCCESSFULLY', false);
    commit('SET_PROFILE_ERROR', false);
  },
};
