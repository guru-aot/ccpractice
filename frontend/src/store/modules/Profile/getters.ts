import { GetterTree } from 'vuex';
import { ProfileState } from './types';
import { RootState } from '../../types';

/**
 * Getters
 */
export const getters: GetterTree<ProfileState, RootState> = {
  /**
   * userProfile
   * @param {*} state
   * @returns  {object} authors
   */
  userProfile(state) {
    return state.currentUser;
  },
  /**
   * successStatus
   * @param {*} state
   * @returns  {object} authors
   */
  successStatus(state) {
    return state.successStatus;
  },
  /**
   * errorStatus
   * @param {*} state
   * @returns  {object} authors
   */
  errorStatus(state) {
    return state.errorStatus;
  },
};
