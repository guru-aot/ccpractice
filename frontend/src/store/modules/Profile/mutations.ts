import { MutationTree } from 'vuex';
import { ProfileState } from './types';

/**
 * mutation
 */
export const mutations: MutationTree<ProfileState> = {
  /**
   * Sets loading
   * @param {*} state
   * @param {boolean} flag - loading
   */

  SET_LOADING(state, flag: boolean) {
    state.loading = flag;
  },
  /**
   * Sets contact
   * @param {*} state
   * @param payload contact list
   */

  // TODO - change to type
  SET_USERS_PROFILE(state, payload: any) {
    state.currentUser = payload;
  },
  SET_PROFILE_SUCCESSFULLY(state, payload: any) {
    // console.log('payload', payload);
    state.successStatus = payload;
  },
  SET_PROFILE_ERROR(state, payload: any) {
    // console.log('payload', payload);
    state.errorStatus = payload;
  },
};
