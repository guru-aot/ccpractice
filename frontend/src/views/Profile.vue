<template>
  <div>
    <Profile />
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import Profile from '@/components/Profile/Profile.vue';
@Component({
  components: {
    Profile
  }
})
export default class ProfilePage extends Vue {}
</script>
