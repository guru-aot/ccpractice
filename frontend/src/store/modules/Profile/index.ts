import { Module } from 'vuex';
import { getters } from './getters';
import { actions } from './actions';
import { mutations } from './mutations';
import { ProfileState } from './types';
import { RootState } from '../../types';

/**
 * Profile state
 */
export const state: ProfileState = {
  currentUser: [],
  loading: false,
  successStatus: false,
  errorStatus: false,
};

const namespaced: boolean = true;

const ProfileModule: Module<ProfileState, RootState> = {
  namespaced,
  state,
  getters,
  actions,
  mutations,
};

export default ProfileModule;
