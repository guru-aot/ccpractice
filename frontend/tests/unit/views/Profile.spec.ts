import { shallowMount, createLocalVue } from '@vue/test-utils';
import Profile from '@/views/Profile.vue';
import Vuex from 'vuex';
import vuetify from 'vuetify';

describe('Profile.vue', () => {
  let wrapper: any;
  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(Vuex);
    localVue.use(vuetify);

    wrapper = shallowMount(Profile, {
      localVue,
    });
  });
  it('renders without crashing', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
