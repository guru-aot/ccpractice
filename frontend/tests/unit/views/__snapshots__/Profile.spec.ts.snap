// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile.vue renders without crashing 1`] = `
<div>
  <profile-stub />
</div>
`;
